File: ui/src/app/shared/jsonrpc/response/addEdgeToUserResponse.ts
import { JsonrpcResponseSuccess } from "../base";

/**
 * Represents a JSON-RPC Response for a {@link AddEdgeToUserRequest}.
 * 
 * <pre>
 * {
 *  "jsonrpc": "2.0",
 *  "id": UUID,
 *  "result": {
 *      "edge": {
 *          id: string,
 *          comment: string,
 *          producttype: string,
 *          version: string,
 *          isOnline: boolean
 *      },
 *      serialNumber: string
 *  }
 * }
 * </pre>
 */
export class AddEdgeToUserResponse extends JsonrpcResponseSuccess {

    public constructor(
        public override readonly id: string,
        public override readonly result: {
            edge: {
                id: string,
                comment: string,
                producttype: string,
                version: string,
                online: boolean
            },
            serialNumber: string
        }
    ) {
        super(id, result);
    }
}

File: io.openems.backend.common/src/io/openems/backend/common/jsonrpc/response/AddEdgeToUserResponse.java
package io.openems.backend.common.jsonrpc.response;

import java.util.UUID;

import com.google.gson.JsonObject;

import io.openems.backend.common.jsonrpc.request.AddEdgeToUserRequest;
import io.openems.backend.common.metadata.Edge;
import io.openems.common.jsonrpc.base.JsonrpcResponseSuccess;
import io.openems.common.utils.JsonUtils;

/**
 * Represents a JSON-RPC Response for {@link AddEdgeToUserRequest}.
 *
 * <pre>
 * {
 *   "jsonrpc": "2.0",
 *   "id": "UUID",
 *   "result": {
 *     "edge": Edge
 *     "serialNumber": String
 *   }
 * }
 * </pre>
 */
public class AddEdgeToUserResponse extends JsonrpcResponseSuccess {

	private final Edge edge;
	private final String serialNumber;

	public AddEdgeToUserResponse(UUID id, Edge edge, String serialNumber) {
		super(id);
		this.edge = edge;
		this.serialNumber = serialNumber;
	}

	@Override
	public JsonObject getResult() {
		return JsonUtils.buildJsonObject() //
				.add("edge", this.edge.toJsonObject()) //
				.addPropertyIfNotNull("serialNumber", this.serialNumber) //
				.build();
	}

}
